src/BlueskyClient/Extensions/PostExtensions.cs

```diff
@@ -6,6 +6,18 @@ namespace BlueskyClient.Extensions;
 
 public static class PostExtensions
 {
+    public static ReadOnlySpan<char> GetRecordKey(this FeedPost post)
+    {
+        var uriSpan = post.Uri.AsSpan();
+        int lastSlashIndex = uriSpan.LastIndexOf('/');
+
+        if (lastSlashIndex != -1)
+        {
+            return uriSpan.Slice(lastSlashIndex + 1);
+        }
+        return null;
```

Comment:
since null can be returned, should the return value be `ReadOnlySpan<char>?`?

Reply:
Hi, unfortunately, it's not possible to define a ReadOnlySpan<char>?  since it's a ref struct. In reality, even though you return a null, once you allocate it, it turns out into an empty string (see test cases)

I have refactored anyway following VS suggestions in order to have something more concrete.

let me know if this looks better

Reply:
I've extracted the logic to a different Helper since I've seen that we require to extract it from different sources, using inmutable classes.